addons/crm_iap_mine/models/crm_iap_lead_mining_request.py

```diff
@@ -333,12 +334,16 @@ def action_get_lead_action(self):
         self.ensure_one()
         action = self.env["ir.actions.actions"]._for_xml_id("crm.crm_lead_all_leads")
         action['domain'] = [('id', 'in', self.lead_ids.ids), ('type', '=', 'lead')]
+        action['context'] = dict(ast.literal_eval(action.get('context', '{}')), create=False)
         return action
 
     def action_get_opportunity_action(self):
         self.ensure_one()
         action = self.env["ir.actions.actions"]._for_xml_id("crm.crm_lead_opportunities")
         action['domain'] = [('id', 'in', self.lead_ids.ids), ('type', '=', 'opportunity')]
+        context = action.get('context', '{}')
+        context = context.replace('uid', str(self.env.uid))
```

Comment:
Is this doing a string replacement? Not great, not sure why we need it as you don't seem to use uid and it should be in the base context.

Reply:
Yes @awa-odoo , this is because of this. Easy to miss, but this override would make the view crash otherwise :bomb: I see this is done for uid in other places.

https://github.com/odoo-dev/odoo/blob/9cf958b9b15a10c622953d879c9861c1d7701097/addons/sale_crm/views/crm_lead_views.xml#L36

Reply:
How about:

`action['context'] = dict(ast.literal_eval(context), create=False, uid=self.env.uid)`

?

A bit cleaner ^^

Reply:
Well actually this would crash because it is the literal_eval function that fails when it encounters` 'default_user_id': uid`, but uid is not known when evaluated. The context is just in its string form, hence the solution.

Also, we would always add this new context value to this action (not only when sale is installed)- I don't know if this is an issue though.

What do you think?

Reply:
Hmm yes I understand the issue, but still not a fan as it could easily fail.

Example:
default_fluid_user: uid would replace the "uid" of "fluid" as well...

If we want to do a string replace, then we probably need to replace the whole key and value to reduce potential issues.

Reply:
Oh I see, I've updated the code and explained the use in the CM.